components/MainUI.tsx

```diff
@@ -1,13 +1,60 @@
-import React from 'react';
-import { Layout } from "./styles";
+import React, { useEffect, useState } from 'react';
+import { Layout } from './styles';
 import { Focus, Bar, Performer } from './';
+import Video from './Video';
 
 const { LayoutContainer } = Layout;
 
 const MainUI = () => {
+  const [participants, setParticipants] = useState([]);
+  const [room, setRoom] = useState(null);
+  const [localParticipant, setLocalParticipant] = useState(null);
+
+  const addParticipant = participant => {
+    console.log('Participant "%s" added', participant.identity);
+    setParticipants(previous => [...previous, participant]);
+  };
+
+  const removeParticipant = participant => {
+    console.log('Participant "%s" REMOVED', participant.identity);
+    setParticipants(previous =>
+      previous.filter(existing => existing.identity !== participant.identity)
+    );
+  };
+
+  const prepareRoom = room => {
+    setRoom(room);
+    setLocalParticipant(room.localParticipant);
+    window.addEventListener('beforeunload', () => room.disconnect());
+    [
+      ...Array.from(room.participants.values()),
+      room.localParticipant
+    ].forEach(participant => addParticipant(participant));
+    room.on('participantConnected', participant => {
```

Comment:
Add a hook to detect participants who join after you. The hierarchy is:

* a room has participants
* participants have tracks (audio, video, and data)